### PR TITLE
Remove new doc flags for restart intent

### PR DESCRIPTION
**Pull request by phillipm**
Strangely, on my particular device (Nexus 5x / Android 6) when I uninstall an app in the app manger the call to `CommCareApplication.restartCommCare` trashes my app, freezing everything until I force close it. 

Removing these flags fix it.
Not really sure what they do / why they were put here in the first place. @ctsims, do you know?

### CODE REVIEW

File: app/src/org/commcare/dalvik/application/CommCareApplication.java [hunk 1]
```diff
@@ -278,11 +278,6 @@ public static void restartCommCare(Activity activity) {
         intent.addFlags(Intent.FLAG_ACTIVITY_CLEAR_TOP |
```
Comment (ctsims) [inline app/src/org/commcare/dalvik/application/CommCareApplication.java]: http://developer.android.com/reference/android/app/Activity.html#finishAndRemoveTask()
http://developer.android.com/reference/android/app/Activity.html#finishAffinity()

Comment (phillipm) [inline app/src/org/commcare/dalvik/application/CommCareApplication.java]: Those require API 16 and API 21. Do you suggest  using them if the API level is met?
Also seems strange to launch an activity from the activity we are closing. Wondering if using a pending intent w/ a timer would be a good approach: http://stackoverflow.com/a/17166729

**Comment by amstone326** [comment]
@phillipm A pending intent approach is what was being used in the app manager's implementation of restart CommCare before I did the unification. Look at what's deleted in SingleAppManagerActivity here: https://github.com/dimagi/commcare-odk/pull/962/files

**Comment by phillipm** [comment]
Just tested out the pending intent stuff and what is currently used seems to work slightly better. 
This is ready to :eyeglasses: again

**Pull request merged**